OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -113,6 +115,17 @@ public void UpdateCell(CPos cell)
 				Dirty.Add(cell + d);
 		}
 
+		void CalculateNetWorth()
```

Comment:
I tried a few more complex and simple things like

```patch
+			NetWorth -= t.Density * type.Info.ValuePerUnit;
+
 			// Set density based on the number of neighboring resources
 			t.Density = ResourceDensityAt(c);
 
+			NetWorth += t.Density * type.Info.ValuePerUnit;
+
```

which will always end in the same result: miscalculations when removing a lone cell without neighbors and unknown previous contents.

As the sum over the whole resource layer is an easy calculation my guess is that things like

```csharp
var previousCells = CVec.Directions.Select(d => d + cell).Where(c => Map.Contains(c));
foreach (var previousCell in previousCells)
{
	var t = Tiles[previousCell];
	if (t.Density > 0)
		NetWorth -= t.Density * t.Type.Info.ValuePerUnit;
}
```

might actually be more costly performance wise.